plutus-core/plutus-core/src/PlutusCore/Parser/ParserCommon.hs

```diff
@@ -48,18 +48,23 @@ intern n = do
             put $ ParserState identifiers'
             return fresh
 
-parse :: Parser a -> String -> T.Text -> Either (ParseErrorBundle T.Text ParserError) a
-parse p file str = runQuote $ parseQuoted p file str
+parse :: ((AsParserErrorBundle e), MonadError e m) =>
+    Parser a -> String -> T.Text -> m a
+parse p file str =
+    throwingEither
+        _ParserErrorBundle
+        (parseToParserErr p file str)
+
+parseToParserErr :: Parser a -> String -> T.Text -> Either ParserErrorBundle a
```

Comment:
I'd probably inline this and just have a direct `throwError` call in the `Left` branch, but not a big deal.

Reply:
Done.